### ISSUE TITLE
iSCSI system extension not working and reported as uninstalled by applications

### ISSUE THREAD

**Issue by Syntax3rror404** [siderolabs/talos#9134]
### Description
I tryed to installed longhorn on Talos.

I installed the following extensions:
```
customization:
  systemExtensions:
    officialExtensions:
      - siderolabs/iscsi-tools
      - siderolabs/util-linux-tools
```

The longhorn system namespace is also privileged
```
pod-security.kubernetes.io/audit: privileged
pod-security.kubernetes.io/enforce: privileged
pod-security.kubernetes.io/warn: privileged
```

And the extensions also showed as installed see "Logs"

But the application shows that the iSCSI tools are not found see "Logs"
In short the following error:
` failed to execute iscsiadm: No such file or directory\n: exit status 127" func=main.main.DaemonCmd.func3 file="daemon.go:92"
`
### Logs
Talos:
```
talosctl get extensions -n 192.168.35.62
NODE            NAMESPACE   TYPE              ID   VERSION   NAME               VERSION
192.168.35.62   runtime     ExtensionStatus   0    1         iscsi-tools        v0.1.4
192.168.35.62   runtime     ExtensionStatus   1    1         util-linux-tools   2.39.3
192.168.35.62   runtime     ExtensionStatus   2    1         schematic          613e1592b2da41ae5e265e8789429f22e121aab91cb4deb6bc3c0b6262961245
```

Longhorn:
```
time="2024-08-09T03:09:40Z" level=fatal msg="Error starting manager: Failed environment check, please make sure you have iscsiadm/open-iscsi installed on the host: failed to execute: /usr/bin/nsenter [nsenter --mount=/host/proc/174187/ns/mnt --net=/host/proc/174187/ns/net iscsiadm --version], output , stderr nsenter: failed to execute iscsiadm: No such file or directory\n: exit status 127" func=main.main.DaemonCmd.func3 file="daemon.go:92"
```

### Environment

- Talos version: 1.7.6
- Kubernetes version: 1.30.2
- Platform: x86_64

**Comment by smira** [comment]
You can check yourself if it it's there or not by using `talosctl ls -r /usr/local` and looking for `iscsi` binaries.

Next, you can use e.g `kubectl debug -it node/<some-node> --image alpine -n kube-system --profile sysadmin` pod to reproduce the issue by executing same command. If it doesn't work, then why?

**Comment by Syntax3rror404** [comment]
Thanks @smira for you're time!

```
talosctl ls -r /usr/local -n 192.168.35.60
NODE            NAME
192.168.35.60   .
192.168.35.60   bin
192.168.35.60   bin/nsenter
192.168.35.60   etc
192.168.35.60   etc/containers
192.168.35.60   etc/containers/iscsid.yaml
192.168.35.60   etc/containers/tgtd.yaml
192.168.35.60   etc/iscsi
192.168.35.60   etc/iscsi/iscsid.conf
192.168.35.60   etc/logrotate.d
192.168.35.60   etc/logrotate.d/iscsiuiolog
192.168.35.60   etc/passwd
192.168.35.60   etc/udev
192.168.35.60   etc/udev/rules.d
192.168.35.60   etc/udev/rules.d/50-iscsi-firmware-login.rules
192.168.35.60   include
192.168.35.60   lib
192.168.35.60   lib/containers
192.168.35.60   lib/containers/iscsid
192.168.35.60   lib/containers/iscsid/dev
192.168.35.60   lib/containers/iscsid/etc
192.168.35.60   lib/containers/iscsid/etc/hosts
192.168.35.60   lib/containers/iscsid/etc/iscsi
192.168.35.60   lib/containers/iscsid/etc/passwd
192.168.35.60   lib/containers/iscsid/etc/resolv.conf
192.168.35.60   lib/containers/iscsid/lib
192.168.35.60   lib/containers/iscsid/proc
192.168.35.60   lib/containers/iscsid/run
192.168.35.60   lib/containers/iscsid/sys
192.168.35.60   lib/containers/iscsid/usr
192.168.35.60   lib/containers/iscsid/usr/lib
192.168.35.60   lib/containers/iscsid/usr/local
192.168.35.60   lib/containers/iscsid/usr/local/lib
192.168.35.60   lib/containers/iscsid/usr/local/sbin
192.168.35.60   lib/containers/iscsid/var
192.168.35.60   lib/containers/iscsid/var/lib
192.168.35.60   lib/containers/iscsid/var/lib/iscsi
192.168.35.60   lib/containers/tgtd
192.168.35.60   lib/containers/tgtd/dev
192.168.35.60   lib/containers/tgtd/etc
192.168.35.60   lib/containers/tgtd/etc/hosts
192.168.35.60   lib/containers/tgtd/etc/resolv.conf
192.168.35.60   lib/containers/tgtd/lib
192.168.35.60   lib/containers/tgtd/proc
192.168.35.60   lib/containers/tgtd/run
192.168.35.60   lib/containers/tgtd/sys
192.168.35.60   lib/containers/tgtd/usr
192.168.35.60   lib/containers/tgtd/usr/lib
192.168.35.60   lib/containers/tgtd/usr/local
192.168.35.60   lib/containers/tgtd/usr/local/sbin
192.168.35.60   lib/containers/tgtd/var
192.168.35.60   lib/containers/tgtd/var/run
192.168.35.60   lib/containers/tgtd/var/run/tgtd
192.168.35.60   lib/kubelet
192.168.35.60   lib/kubelet/credentialproviders
192.168.35.60   lib/libisns.so
192.168.35.60   lib/libisns.so.0
192.168.35.60   lib/libopeniscsiusr.so
192.168.35.60   lib/libopeniscsiusr.so.0
192.168.35.60   lib/libopeniscsiusr.so.0.2.0
192.168.35.60   lib/pkgconfig
192.168.35.60   lib/pkgconfig/libopeniscsiusr.pc
192.168.35.60   sbin
192.168.35.60   sbin/brcm_iscsiuio
192.168.35.60   sbin/fstrim
192.168.35.60   sbin/iscsi-gen-initiatorname
192.168.35.60   sbin/iscsi-iname
192.168.35.60   sbin/iscsi_discovery
192.168.35.60   sbin/iscsi_fw_login
192.168.35.60   sbin/iscsi_offload
192.168.35.60   sbin/iscsiadm
192.168.35.60   sbin/iscsid
192.168.35.60   sbin/iscsid-wrapper
192.168.35.60   sbin/iscsistart
192.168.35.60   sbin/iscsiuio
192.168.35.60   sbin/tgtadm
192.168.35.60   sbin/tgtd
192.168.35.60   sbin/tgtimg
192.168.35.60   share
192.168.35.60   share/ca-certificates
192.168.35.60   share/schematic
192.168.35.60   share/schematic/613e1592b2da41ae5e265e8789429f22e121aab91cb4deb6bc3c0b6262961245
```

```
kubectl debug -it node/talos-670-ud0 --image alpine -n kube-system
Creating debugging pod node-debugger-talos-670-ud0-rbgzm with container debugger on node talos-670-ud0.
If you don't see a command prompt, try pressing enter.
/ # iscsiadm
/bin/sh: iscsiadm: not found
```

**Comment by smira** [comment]
You won't be able to run `iscsiadm` directly from the pod, as the pod filesystem is not the host filesystem.

You can try `nsenter` entering PID1 mount namespace, or mimic whatever your workload tries to do.

**Comment by Syntax3rror404** [comment]
@smira 
Thank you again for your time! :) 

Ok ...... Just redeployed longhorn with helm and now its working. Not sure whats going on and whats causing this error in the last tryes. 

I installed the extension via upgrade

`talosctl upgrade --image factory.talos.dev/installer/613e1592b2da41ae5e265e8789429f22e121aab91cb4deb6bc3c0b6262961245:v1.7.6 --nodes 192.168.35.61 --preserve`

I read about some deprecations. Is this the deprecated method? The documentation is not clear about this. 

Second thing is I applyed this patch to all the nodes but seams like the mounts cannot be upgraded. Is this expected? I cant reinstalled talos every time I need to patch the disks and/or the mounts? Not sure what Im missing here
```
machine:
    kubelet:
        extraMounts:
        - destination: /var/lib/longhorn
          type: bind
          source: /var/lib/longhorn
          options:
          - bind
          - rshared
          - rw
```

**Comment by smira** [comment]
`talosctl mounts` shows host mounts only, so it won't show this mount (as it's in a container).

Using `--preserve` option is fine, and it will be a default in Talos 1.8+.

**Comment by Syntax3rror404** [comment]
Hello @smira again thanks for you're answer :) 

I understand!!!

Is it bad practise to install system extrension over talosctl upgrade --image ? The documantation tells about a deprecation, but I dont find a alternate approach to patch a system.

If I use a second disk for data and I reset the system or not using the --preserve arg gets the ! system disk also wiped? If not it would be possible to not use the preserve and holding the system clean without wiping the relevant data?

**Comment by smira** [comment]
The [guide](https://www.talos.dev/v1.7/talos-guides/install/boot-assets/) linked in the documentation you posted provides a better way to install extensions.

tl;dr the easiest is to use [Image Factory](https://factory.talos.dev/?arch=amd64&cmdline-set=true&extensions=-&extensions=siderolabs%2Fiscsi-tools&platform=metal&target=metal&version=1.7.6): this link provides an amd64 bare-metal image with iscsi extension (but you can pick your own version)
